components/nimbus/src/enrollment.rs

```diff
@@ -670,6 +671,16 @@ impl<'a> EnrollmentsEvolver<'a> {
                 }
             };
 
+            #[cfg(feature = "stateful")]
+            if let Some(enrollment) = next_enrollment.clone() {
```

Comment:
Why are we cloning here? You can do `if let Some(ref enrollment) = next_enrollment` and `enrollment` will be `&ExperimentEnrollment`.